### PR TITLE
[feature] Squash Rz gates in `globalise_PhasedX`

### PR DESCRIPTION
**Pull request by lmondada**

### CODE REVIEW

File: pytket/docs/changelog.rst [hunk 1]
```diff
@@ -12,6 +12,7 @@ Minor new features:
   that can be rendered by the ``graphviz`` package.
 * New pass and transform `DecomposeTK2` to decompose TK2 gates using the
   approximate KAK decomposition.
+* ``GlobalisePhasedX`` uses fewer Rz rotations.
```
Comment (cqc-melf) [inline pytket/docs/changelog.rst:15]: Maybe clarify that this is a compiler pass?

File: tket/src/Transformations/BasicOptimisation.cpp [hunk 2]
```diff
@@ -580,6 +580,132 @@ Transform commute_SQ_gates_through_SWAPS(const op_node_errors_t &node_errors) {
...
+          for (unsigned i = 0; i < arity; ++i) {
+            Vertex in_v = circ.source(in_edges[i]);
+            Op_ptr in_op = circ.get_Op_ptr_from_Vertex(in_v);
+            Vertex out_v = circ.target(out_edges[i]);
+            Op_ptr out_op = circ.get_Op_ptr_from_Vertex(out_v);
+
+            Expr angle;
+            Edge in_e, out_e;
+            VertexSet bin;
+            if (in_op->get_type() == OpType::Rz) {
```
Comment (cqc-melf) [inline tket/src/Transformations/BasicOptimisation.cpp:661]: Would it be possible to add the condition that all RZ after executing this pass will be in front or after the NPhasedX?

**Comment by cqc-melf** [comment]
Only two short comments, looks good!

File: pytket/docs/changelog.rst (hunk 1 above)
Comment (lmondada) [reply pytket/docs/changelog.rst:15]: Done!

File: tket/src/Transformations/BasicOptimisation.cpp (hunk 2 above)
Comment (cqc-melf) [reply tket/src/Transformations/BasicOptimisation.cpp:661]: Yes, thank you for explaining that!

**Review by cqc-melf — APPROVED** [review]

**Pull request merged**